internal/commands/command.go

```diff
@@ -95,6 +100,11 @@ func BuildCommand(child Command, parent *cobra.Command, config *config.Config) C
 
 	// Set run
 	child.Cobra().RunE = func(cmd *cobra.Command, args []string) error {
+		// Do not create service for offline commands, e.g. upctl version
+		if i, ok := child.(OfflineCommand); ok && i.IsOffline() {
```

Comment:
Would there be a better way to tell `BuildCommand()` that some commands do not need services connection than this interface check?

Reply:
This seems fine to me, it's not a completely foreign pattern and other options would need much more changes (lazy init of service? move the service init to `commandRunE` and make it a separate type of a command? something else?).

Although, commonly this pattern has only a member method with no return value (eg `type OfflineCommand interface { IsOffline() }`) as this prevents the confusing situation where something fulfills `OnlineCommand` but the method returns false. That is, using the fulfillment of the interface as the definition rather than what that particular method returns. If you go that route, maybe a good idea to make the name of the method a bit more explicit too - just `IsOffline` doesn't really say much unless one knows the context.

Reply:
I removed the return value and renamed `IsOffline` to `DoesNotUseServices`.

I also though of separate type of command, but that seemed like a lot of work as we probably wont have much commands that don't need the API.

Reply:
That works great :+1: